### PR TITLE
Add GetUnhandledSignalNames

### PR DESCRIPTION
**Pull request by longquanzheng**
## What was changed

Expose the API `GetUnhandledSignalNames`
## Why?

See https://temporalio.slack.com/archives/CTDTU3J4T/p1672780768847019?thread_ts=1672770053.462709&cid=CTDTU3J4T

## Checklist

1. Closes 
None. but related to https://github.com/temporalio/sdk-features/issues/51 

2. How was this tested:

There are already tests covered for this API internally in the SDK.

3. Any docs updates needed?

Maybe worth mentioning for using continueAsNew, this API will be helpful to check if all the signals are drained.

### CODE REVIEW

File: workflow/workflow.go [hunk 1]
```diff
@@ -197,6 +197,11 @@ func GetMetricsHandler(ctx Context) metrics.Handler {
 	return internal.GetMetricsHandler(ctx)
 }
 
+// GetUnhandledSignalNames returns signal names that have not been consumed.
```
Comment (cretz) [inline workflow/workflow.go:200]: But they may have been consumed, just not consumed completely. More accurately:

```suggestion
// GetUnhandledSignalNames returns signal names that have unconsumed signals.
```

**Review by cretz — APPROVED** [review]
LGTM. Minor nit.

**Comment by cretz** [comment]
@longquanzheng - sorry, some of the PRs are moving fast on this repo. Can you update your branch w/ latest master (or let me know and I can)?

**Comment by longquanzheng** [comment]
> @longquanzheng - sorry, some of the PRs are moving fast on this repo. Can you update your branch w/ latest master (or let me know and I can)?

no worries, I can do it. You are also welcome to do it anytiime. LMK if I need to do anything to change permission

**Pull request merged**